### PR TITLE
internal: move backoff to internal

### PR DESCRIPTION
**Pull request by menghanl**
So other components such as grpclb can reuse the backoff implementation.

### CODE REVIEW

File: internal/backoff/backoff.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	// the number of consecutive failures.
+	Backoff(retries int) time.Duration
+}
+
+// Config defines the parameters for the default gRPC backoff strategy.
+type Config struct {
+	// MaxDelay is the upper bound of backoff delay.
+	MaxDelay time.Duration
+
+	// TODO(stevvooe): The following fields are not exported, as allowing
```
Comment (dfawley) [inline internal/backoff/backoff.go]: The following fields *are* exported.  :)

Should the TODO go back to where it was before?

File: internal/backoff/backoff.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package backoff
```
[the remark this answers is not in the document]
Comment (menghanl) [reply internal/backoff/backoff.go]: Done

File: internal/backoff/backoff.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import (
+	"time"
+
+	"google.golang.org/grpc/internal/grpcrand"
+)
+
+// Strategy defines the methodology for backing off after a grpc connection
+// failure.
+//
+// This is kept in internal until the gRPC project decides whether or not to
```
[the remark this answers is not in the document]
Comment (menghanl) [reply internal/backoff/backoff.go]: Moved.

File: internal/backoff/backoff.go (hunk 1 above)
Comment (menghanl) [reply internal/backoff/backoff.go]: Deleted this TODO.

**Pull request merged**